crates/ln-dlc-node/src/node.rs

```diff
@@ -259,21 +267,93 @@ impl Node {
         let (height, header) = ln_dlc_wallet.tip().unwrap();
         let hash = header.block_hash();
 
+        // Read ChannelMonitor state from disk
+        let mut channelmonitors = persister
+            .read_channelmonitors(keys_manager.clone())
+            .unwrap();
+
         let channel_manager = {
-            let chain_params = ChainParameters {
-                network,
-                best_block: BestBlock::new(hash, height),
-            };
-            Arc::new(ChannelManager::new(
-                ln_dlc_wallet.clone(),
-                chain_monitor.clone(),
-                ln_dlc_wallet.clone(),
-                logger.clone(),
-                keys_manager.clone(),
-                ldk_user_config,
-                chain_params,
-            ))
+            if let Ok(mut f) = std::fs::File::open(format!("{ldk_data_dir}/manager")) {
+                let mut channel_monitor_mut_references = Vec::new();
+                for (_, channel_monitor) in channelmonitors.iter_mut() {
+                    channel_monitor_mut_references.push(channel_monitor);
+                }
+                let read_args = ChannelManagerReadArgs::new(
+                    keys_manager.clone(),
+                    ln_dlc_wallet.clone(),
+                    chain_monitor.clone(),
+                    ln_dlc_wallet.clone(),
+                    logger.clone(),
+                    ldk_user_config,
+                    channel_monitor_mut_references,
+                );
+                <(BlockHash, ChannelManager)>::read(&mut f, read_args)
+                    .map_err(|e| anyhow::anyhow!("{e:?}"))
```

Comment:
Given you unwrap here I don't think it is useful to map the error to anyhow.

Reply:
this needs to be done as the trait bound `lightning::ln::msgs::DecodeError: std::error::Error` is not satisfied. Mapping the error to an anyhow error allows me to use the `?`. Any suggestions on how this can be done better?

Reply:
It was only because you unwraped right aftewards. In a later commit you changed it to `?`, so this comment is actually outdated :)